File: unidbg-android/src/main/java/com/vortexdbg/linux/struct/IFConf.java
package com.vortexdbg.linux.struct;

import com.vortexdbg.Emulator;
import com.vortexdbg.pointer.UnidbgStructure;
import com.sun.jna.Pointer;

public abstract class IFConf extends UnidbgStructure {

    public static IFConf create(Emulator<?> emulator, Pointer pointer) {
        return emulator.is64Bit() ? new IFConf64(pointer) : new IFConf32(pointer);
    }

    public int ifc_len;

    public IFConf(Pointer p) {
        super(p);
    }

    public abstract long getIfcuReq();

}

File: vortexdbg-android/src/main/kotlin/com/vortexdbg/linux/struct/IFConf.kt
package com.vortexdbg.linux.struct

import com.vortexdbg.Emulator
import com.vortexdbg.pointer.VortexdbgStructure
import com.sun.jna.Pointer

abstract class IFConf(p: Pointer?) : VortexdbgStructure(p) {

    @JvmField
    var ifc_len: Int = 0

    abstract fun getIfcuReq(): Long

    companion object {
        @JvmStatic
        fun create(emulator: Emulator<*>, pointer: Pointer?): IFConf {
            return if (emulator.is64Bit()) IFConf64(pointer) else IFConf32(pointer)
        }
    }

}
